### PR TITLE
Adds auto labelling of changes to the SQL, icons, and sounds trees. Also autolabels reverts

### PR DESCRIPTION
**Pull request by Cyberboss**
@MrStonedOne

### CODE REVIEW

**Comment by RemieRichards** [comment]
Is this an outdated version of the tool? it doesn't have 'Needs Pictures' which the active bot clearly does.

**Comment by Cyberboss** [comment]
I literally don't know why it touched the Needs Pictures label, because this is the code the bot is running

**Comment by RemieRichards** [comment]
Well, add that anyway.
or atleast add the 'Sprites' tag when the /icon directory is touched.

**Comment by Cyberboss** [comment]
Oh ik what it is: Someone added the Need Pictures label while the bot was processing a request and when the bot does that it sets the array of labels to whatever it was when it got the request +/- anything it modifies itself, which wouldn't include Needs Pictures

**Comment by RemieRichards** [comment]
Silly robot.

**Comment by RemieRichards** [comment]
How about a DNM, WIP, style one for REVERT/REMOVAL?

**Comment by Cyberboss** [comment]
We already have those though? https://github.com/tgstation/tgstation/pull/28847/files#diff-4daa66d50fe74a3d9ff8450404f01504R163 What do you mean for Revert/Removal?

**Comment by RemieRichards** [comment]
DNM, WIP style, as in a [WORD] style one. I figured it was p.self explanatory.

**Comment by RemieRichards** [comment]
fix, fixes?

**Comment by Cyberboss** [comment]
That ones a bit to general imo, we already add the fix tag if the PR closes an issue

**Comment by RemieRichards** [comment]
we as in humanity, or we as in the robot?

**Comment by Cyberboss** [comment]
Depends, who bears responsibility when the robot becomes sentient?

**Comment by RemieRichards** [comment]
the robot does.

File: tools/github_webhook_processor.php [hunk 1]
```diff
@@ -146,15 +157,11 @@ function tag_pr($payload, $opened) {
...
-	if(has_tree_been_edited($payload, '_maps'))
-		$tags[] = 'Map Edit';
-	else
-		$remove[] = 'Map Edit';
-
-	if(has_tree_been_edited($payload, 'tools'))
-		$tags[] = 'Tools';
-	else
-		$remove[] = 'Tools';
+	autotag_tree($payload, '_maps', 'Map Edit', $tags, $remove);
```
Comment (lzimann) [inline tools/github_webhook_processor.php]: Isn't there a way to make this a dictionary or something so there's no copypasta?

**Comment by MrStonedOne** [comment]
Installed.

Wait a bit to see if it's working and then merge.

**Pull request merged**